### ISSUE TITLE
Consistent behaviour for moderation for Public->Private->Public stories

### ISSUE THREAD

**Issue by helendduncan** [alan-turing-institute/AutSPACEs#445]
### Summary

As noted by @gedankenstuecke. When flipping an existing experience from public to private and back, the moderation page crashes in a way that was observed when there were PublicExperience objects that did not have a UUID.

### Deadline 
Feb 2023
### What needs to be done?

- [ ] Investigate the cause
- [ ] Modify code to fix
- [ ] Additional: Think of tests that we can apply to check for expected behaviour

### Who can help?
@helendduncan 

---

**Comment by gedankenstuecke** [comment]
Okay, I think I found out what causes the issue: 

When creating a new experience we call the `update_public_experience_db` function like this in the `share_experience` view: https://github.com/alan-turing-institute/AutSPACEs/blob/267a7aef5e34b96c6f1d08e30444c66132b8df9a/server/apps/main/views.py#L76

As part of this we set the `uuid=uuid` in the function call. The `update_public_experience_db` has a parameter, but then the actual creation of the public experience is the following: 

https://github.com/alan-turing-institute/AutSPACEs/blob/267a7aef5e34b96c6f1d08e30444c66132b8df9a/server/apps/main/views.py#L129

We don't explicitly set the different model parameters from the `PublicExperience` model, but rather use the `data` dict which now includes the `uuid`. The problem is that the `PublicExperience` model still expects the key to be called `experience_id` (see below). 

https://github.com/alan-turing-institute/AutSPACEs/blob/267a7aef5e34b96c6f1d08e30444c66132b8df9a/server/apps/main/models.py#L10

As a result the `experience_id` field always remains empty. This breaks the moderation form, but also leads to the problem that we currently can only store a single public experience as the `experience_id` is the primary key. 😂 

**Potential solutions**: 
1. rename the `experience_id` in the model to `uuid`, then the current calls should work 
2. explicitly rename the `uuid` to `experience_id` in the function call should also work
